modules/install-upgrade/pages/upgrade-guide.adoc

```diff
@@ -0,0 +1,554 @@
+
+// known issues
+
+=== Kubernetes deployment using Helm chart
+
+The Helm chart for Luna Streaming is available in the https://github.com/datastax/pulsar-helm-chart/blob/master/helm-chart-sources/pulsar/values.yaml[Helm chart sources] repository.
+
+Deploying Luna Streaming on Kubernetes using the DataStax Helm chart is another common method for running Pulsar in a cloud-native environment.
+
+. Back up your existing Pulsar data and configurations to prevent data loss.
+. To save your current Helm release configuration, run the following command:
++
+[source,bash]
+----
+helm get values > pulsar-backup-values.yaml
+----
++
+. To update the DataStax Pulsar Helm chart repository, run the following command:
++
+[source,bash]
+----
+helm repo update
+----
++
+. Open `helm-chart-sources/pulsar/values.yaml` and update the image tag to 3.1.0 (or the specific tag you wish to use).
++
+[source,yaml]
+----
+image:
+  broker:
+    # If not using tiered storage, you can use the smaller pulsar image for the broker
+    repository: datastax/lunastreaming-all
+    pullPolicy: IfNotPresent
+    tag: 3.1_4.5
+  brokerSts:
+    # If not using tiered storage, you can use the smaller pulsar image for the broker
+    repository: apachepulsar/pulsar
+    pullPolicy: IfNotPresent
+    tag: latest
+  function:
+    repository: apachepulsar/pulsar
+    pullPolicy: IfNotPresent
+    tag: latest
+  zookeeper:
+    repository: apachepulsar/pulsar
+    pullPolicy: IfNotPresent
+    tag: latestupgr
+  bookkeeper:
+    repository: apachepulsar/pulsar
+    pullPolicy: IfNotPresent
+    tag: latest
+  proxy:
+    repository: apachepulsar/pulsar
+    pullPolicy: IfNotPresent
+    tag: latest
+  bastion:
+    repository: apachepulsar/pulsar
+    pullPolicy: IfNotPresent
+    tag: latest
+----
+. Review and modify any other configuration parameters that may have changed between versions, such as resource limits, storage classes, and additional components. To modify other configurations, update `values.yaml` as needed. For example, to modify the broker's replica count, update the following fields:
++
+[source,yaml]
+----
+broker:
+  component: broker
+  replicaCount: 2
+  configData:
+    brokerDeduplicationEnabled: "false"
+----
++
+. To upgrade your existing Pulsar installation, run the following Helm command.
+This command assumes the default `pulsar` namespace. If you are using a different namespace, replace `pulsar` with your namespace.
+The `--wait` flag ensures that Helm waits until all pods are ready before completing the upgrade.
++
+[source,bash]
+----
+helm upgrade pulsar datastax-pulsar/pulsar --namespace pulsar --values current-values.yaml --wait
+----
++
+. To check the status of the pods to ensure they are running correctly, run the following command:
++
+[source,bash]
+----
+kubectl get pods --namespace pulsar
+----
+=
+. To check the logs for any issues, run the following command:
++
+[source,bash]
+----
+kubectl logs -n
```

Comment:
`kubectl logs` command is not correct. Please update with below given command:
`kubectl logs <pod-name> -n <namespace>`